File: examples/vercel-ai-service/src/hot-path.ts
import { config as loadEnv } from "dotenv"
import { dirname, resolve } from "node:path"
import { fileURLToPath } from "node:url"
import { MemexAI } from "@memexai/sdk"

const here = dirname(fileURLToPath(import.meta.url))
loadEnv({ path: resolve(here, "../../../.env"), quiet: true })
loadEnv({ path: resolve(here, "../.env"), override: true, quiet: true })

const memex = new MemexAI({
  url: process.env["MEMEX_URL"] ?? "http://localhost:8080",
  apiKey: process.env["MEMEX_API_KEY"] ?? "dev-agent-key",
})
const memory = memex.forUser({
  userId: process.env["MEMEX_USER_ID"] ?? "example_vercel_ai_hot_path_user",
  actor: "vercel-ai-hot-path",
})

const userMessage = "Remember that I prefer ceramic pour-over coffee with oat milk."
const result = await memory.remember({ text: userMessage })

console.log("hot_path: awaited memory.remember before finishing the turn")
console.log(`writes: ${result.writes.map((write) => write.path).join(", ") || "none"}`)

File: examples/llamaindex-python/hot_path.py
import asyncio
import os

from dotenv import load_dotenv
from memexai import MemexAI


async def main() -> None:
    load_dotenv()
    memex = MemexAI(url=os.getenv("MEMEX_URL", "http://localhost:8080"), api_key=os.getenv("MEMEX_API_KEY", "dev-agent-key"))
    try:
        memory = memex.for_user(os.getenv("MEMEX_USER_ID", "llamaindex_hot_path_user"), actor="llamaindex-hot-path")
        result = await memory.remember("Remember that I prefer LlamaIndex workflow steps for memory extraction.")
        print("hot_path: await memory.remember in the current workflow step")
        print(result.get("writes", []))
    finally:
        await memex.close()


if __name__ == "__main__":
    asyncio.run(main())
